docs/cookbook/creating_custom_rest_api_response.md

```diff
@@ -0,0 +1,268 @@
+#Creating custom REST API response based on Accept header
+
+There are several situations when customizing REST API response could be useful. One of the examples is extending it by adding comments count to 
+`eZ\Publish\Core\REST\Server\Output\ValueObjectVisitor\VersionInfo` response.
+
+##Implementation of dedicated Visitor
+The first step is creating your own implementation of `ValueObjectVisitor`. It contains all the logic which is responsible for fetching data
+you want to present and for modifying the actual response.
+
+```php
+<?php
+
+namespace Acme\ExampleBundle\Rest\ValueObjectVisitor;
+
+use eZ\Publish\API\Repository\Repository;
+use eZ\Publish\Core\REST\Common\Output\Generator;
+use eZ\Publish\Core\REST\Common\Output\Visitor;
+use eZ\Publish\Core\REST\Server\Output\ValueObjectVisitor\VersionInfo as BaseVersionInfo;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo as APIVersionInfo;
+
+class VersionInfo extends BaseVersionInfo
+{
+    private $repository;
+
+    public function __construct(Repository $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    protected function visitVersionInfoAttributes(Visitor $visitor, Generator $generator, APIVersionInfo $versionInfo)
+    {
+        parent::visitVersionInfoAttributes($visitor, $generator, $versionInfo);
+
+        $this->visitTestValue($generator);
+    }
+
+    protected function visitTestValue(Generator $generator)
+    {
+        $commentsCount = 10; //for this example value is mocked, but you can use any custom logic to fetch it
+        
+        $generator->startValueElement('commentsCount', $commentsCount);
+        $generator->endValueElement('commentsCount');
+    }
+}
+```
+
+##Overriding response type
+Next, you'll need to make sure that your new implementation of serialization applies only to selected objects. In order to do that you need to 
+decorate `eZ\Publish\Core\REST\Common\Output\ValueObjectVisitorDispatcher` from `ezpublish-kernel`.
+
+```php
+<?php
+
+namespace Acme\ExampleBundle\Rest;
+
+use eZ\Publish\Core\REST\Common\Output\Generator;
+use eZ\Publish\Core\REST\Common\Output\ValueObjectVisitorDispatcher as BaseValueObjectVisitorDispatcher;
+use eZ\Publish\Core\REST\Common\Output\Exceptions\NoVisitorFoundException;
+use eZ\Publish\Core\REST\Common\Output\Visitor;
+
+class ValueObjectVisitorDispatcher extends BaseValueObjectVisitorDispatcher
+{
+    private $parentDispatcher;
+
+    public function __construct(BaseValueObjectVisitorDispatcher $parentDispatcher)
+    {
+        $this->parentDispatcher = $parentDispatcher;
+    }
+
+    public function setOutputVisitor(Visitor $outputVisitor)
```

Comment:
Not a big fan of these, but I guess that you couldn't find a simpler way to get this to work ?

Reply:
Improvment can be done as a followup, given the doc here builds upon what has been released, it's imho good to go for now.

Maybe @adamwojs and/or @bdunogier can create a feature request to handle the improvment once there is priority for it.